src/Microsoft.Azure.EventHubs/Amqp/AmqpExceptionHelper.cs

```diff
@@ -75,7 +77,8 @@ public static Exception ToMessagingContract(string condition, string message, bo
             }
             else if (string.Equals(condition, AmqpErrorCode.NotFound.Value))
             {
-                if (message.ToLower().Contains("status-code: 404"))
+                if (message.ToLower().Contains("status-code: 404") ||
+                    Regex.IsMatch(message, "The messaging entity .* could not be found"))
```

Comment:
isMatch call will compile the expression once and cache. Not really needed to cache a Regex instance here. Besides this is on the exception handling code path, no perf impact event not cached.

---
In reply to: [83526837](https://github.com/Azure/azure-event-hubs-dotnet/pull/22#discussion_r83526837) [](ancestors = 83526837)